public/js/index.js

```diff
@@ -1,57 +0,0 @@
-(()=>{
-  dom();
-})();
-
-function dom() {
-  if (typeof document != 'undefined') {
-    var searchForm = document.getElementById('searchForm');
-    if (searchForm) {
-      searchForm.addEventListener('submit', function(event){
-        event.preventDefault();
-        var userName = event.target.firstElementChild.value;
-        apiReq(userName , function(data){
-          document.querySelector('.avatar img').setAttribute("src", data.avatar_url);
-          document.querySelector('.names h1').textContent = data.name;
-          document.querySelector('.names h6').textContent = data.login;
-          document.querySelector('#company').textContent = data.company;
-          document.querySelector('#location').textContent = data.location;
-          document.querySelector('#followersNumber span').textContent = data.followers;
-          document.querySelector('#followingNumber span').textContent = data.following;
-          document.querySelector('.listRepos').innerHTML = myRepos(data.repos);
-        });
-  	  });
-    }
-  }
-}
-
-// Generate repos html as string
-function myRepos(reposArray) {
-  //return string with html needed to display repos
-    var result = '';
-    reposArray.map(function(repo) {
-      result += '<div class="repoData">'+
-                  '<a class="repoName" href="https://www.github.com/'+repo.name+'" class="repoName" alt="'+repo.name+'" target="_blank">'+repo.name+'</a>'+
-                    '<p class="repoDesc">'+repo.desc+'</p>'+
-                    '<p class="repoLang">Language : <span>'+repo.lang+'</span></p>'+
-              '</div>';
-    });
-    return result;
-  }
```

Comment:
This is a very bad use of map. Map is for _returning a new array_. You are using it like a `forEach` but really you should use a reduce to change an array into something else (a string, in this case):
```
// Generate repos html as string
function myRepos(reposArray) {
  //return string with html needed to display repos
  return reposArray.reduce(function(result, repo) {
      return result += '<div class="repoData">'+
                '<a class="repoName" href="https://www.github.com/'+repo.name+'" class="repoName" alt="'+repo.name+'" target="_blank">'+repo.name+'</a>'+
                  '<p class="repoDesc">'+repo.desc+'</p>'+
                  '<p class="repoLang">Language : <span>'+repo.lang+'</span></p>'+
            '</div>';
    },
    ''
  );
}
```